patches/chromium/reland_sequentialise_access_to_callbacks_in.patch

```diff
@@ -0,0 +1,223 @@
+separate threads, there may be race conditions around the
+pending_callbacks_ member of DWriteFontLookupTableBuilder.
+
+Sequentialise and guard access to pending_callbacks_ with a separate
+sequenced task runner.
+
+Remove explicit configuration of the FontUniqueNameBrowserTest cache
+directory as [1] introduced a callback function in
+ShellContentBrowserClient which sets this correctly. This avoids
+instantiating DWriteFontLookupTableBuilder too early when the ThreadPool
+is not yet available in a BrowserTest.
+
+[1] https://chromium-review.googlesource.com/c/chromium/src/+/1776358/9/content/shell/browser/shell_content_browser_client.cc#422
+
+Fixed: 1047054
+Change-Id: I38cf8b84a48315980624b68bbf55d3727be457b8
+Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/2032119
+Reviewed-by: Matthew Denton <mpdenton@chromium.org>
+Commit-Queue: Dominik Röttsches <drott@chromium.org>
+Cr-Commit-Position: refs/heads/master@{#737466}
+
+diff --git a/content/browser/font_unique_name_lookup/font_unique_name_browsertest.cc b/content/browser/font_unique_name_lookup/font_unique_name_browsertest.cc
+index 18e63dfaec08f37e107215876f4a0511542148d8..8266e6fc1628310dafd497debbe1c6ba43aa02ca 100644
+--- a/content/browser/font_unique_name_lookup/font_unique_name_browsertest.cc
++++ b/content/browser/font_unique_name_lookup/font_unique_name_browsertest.cc
+@@ -119,20 +119,10 @@ class FontUniqueNameBrowserTest : public DevToolsProtocolTest {
+   }
+ 
+ #if defined(OS_WIN)
+-  // The Windows service for font unique name lookup needs a cache directory to
+-  // persist the cached information. Configure a temporary one before running
+-  // this test.
+-  void SetUpInProcessBrowserTestFixture() override {
+-    DevToolsProtocolTest::SetUpInProcessBrowserTestFixture();
+-    DWriteFontLookupTableBuilder* table_builder =
+-        DWriteFontLookupTableBuilder::GetInstance();
+-    ASSERT_TRUE(cache_directory_.CreateUniqueTempDir());
+-    table_builder->SetCacheDirectoryForTesting(cache_directory_.GetPath());
+-  }
+-
+   void PreRunTestOnMainThread() override {
+     DWriteFontLookupTableBuilder* table_builder =
+         DWriteFontLookupTableBuilder::GetInstance();
++    table_builder->ResetStateForTesting();
+     table_builder->SchedulePrepareFontUniqueNameTableIfNeeded();
+     DevToolsProtocolTest::PreRunTestOnMainThread();
+   }
+diff --git a/content/browser/renderer_host/dwrite_font_lookup_table_builder_win.cc b/content/browser/renderer_host/dwrite_font_lookup_table_builder_win.cc
+index f8ba10769691864f59ea6f2ef55700d0cd9c4969..c5beb1fb980c0b42e00508e09179c83dfe48e9cf 100644
+--- a/content/browser/renderer_host/dwrite_font_lookup_table_builder_win.cc
++++ b/content/browser/renderer_host/dwrite_font_lookup_table_builder_win.cc
+@@ -23,6 +23,7 @@
+ #include "base/strings/utf_string_conversions.h"
+ #include "base/task/post_task.h"
+ #include "base/task/task_traits.h"
++#include "base/task/thread_pool.h"
```

Comment:
`thread_pool` api is not available in this version. `base::ThreadPool::CreateSequencedTaskRunner`

Reply:
rats.